pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -36,12 +37,12 @@ func Add(mgr manager.Manager) error {
 }
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+func newReconciler(mgr manager.Manager) *ReconcileCheInstallation {
 	return &ReconcileCheInstallation{client: mgr.GetClient(), scheme: mgr.GetScheme()}
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
+func add(mgr manager.Manager, r *ReconcileCheInstallation) error {
```

Comment:
same here: why do you want to change the param type?

Reply:
To make this statement compile: https://github.com/nurali-techie/toolchain-operator/pull/1/files?file-filters%5B%5D=.go#diff-b09f4db8198fe6aeba0154bd7f8fdfbeR76

Reply:
ok, got it.

Reply:
because you need to have the references to the `ReconcileCheInstallation` to be able to assign `watchCheCluster func() error`